### DISCUSSION TITLE
Auto recover VPN connection after temporary internet disconnection

### DISCUSSION THREAD

**Discussion by smbdar** [strongswan/strongswan#1075] [Q&A] [answered]
Hello,

to keep VPN tunnel alive I set /etc/ipsec.conf this way:

```
conn someVPN
    auto=route
    keyexchange=ikev2
#
    leftsourceip=%config4
    leftcert=someUser_Cert.der
#
    right=somevpn.com.pl
    rightid=@somevpn.com.pl
    rightsubnet=10.215.121.0/24
```

Created a new cron job:

`*/15 * * * * ping -c 2 10.215.121.100`

And set "charon.install_virtual_ip_on = lo" in /etc/strongswan.conf.

The point of changes is to automatically reconnect to VPN server after temporarily losing internet connection for whatever reason (WiFi disconnection, router reboot, power shortage, ISP maintenance, etc.). 

Is there a better way to achieve this goal?

**Comment by tobiasbrunner** [comment, marked as answer]
I guess this is the safest way to ensure that. While there is the combination of `auto=start`, `keyingtries=%forever` and `close|dpdaction=restart`, there are currently situations where an SA is still terminated (e.g. after a fatal error like NO_PROPOSAL_CHOSEN). With the trap policy and forced traffic the SA is automatically recreated even in such situations.

Note that if you are not using the latest release (5.9.6), you might want to disable reauthentication (`reauth=no`, although the server might still request a reauthentication if it has not disabled it on its end) or use make-before-break reauthentication (`charon.make_before_break`) as the combination of trap policies and break-before-make reauthenticaiton could lead to duplicate SAs (see [here](https://docs.strongswan.org/docs/5.9/config/rekeying.html) for details).

**Reply by smbdar** [reply]
A little follow-up: after resume notebook wouldn't connect to a VPN server, so I created an executable script "/etc/network/if-up.d/ipsec" with content:

```
#!/bin/sh
echo `date +%Y-%m-%d\ %H:%M:%S`' ipsec restart on network connection' >> /tmp/keep_ipsec.log
ipsec restart
exit 0
```

It gets called 2 times after WiFi or cable connection, but at least it works :-)
